qaframework-bdd-tests/src/main/java/org/openmrs/contrib/qaframework/page/EncounterFormPage.java

```diff
@@ -0,0 +1,84 @@
+/**
+ * This Source Code Form is subject to the terms of the Mozilla Public License,
+ * v. 2.0. If a copy of the MPL was not distributed with this file, You can
+ * obtain one at http://mozilla.org/MPL/2.0/. OpenMRS is also distributed under
+ * the terms of the Healthcare Disclaimer located at http://openmrs.org/license.
+ *
+ * Copyright (C) OpenMRS Inc. OpenMRS is a registered trademark and the OpenMRS
+ * graphic logo is a trademark of OpenMRS Inc.
+ */
+package org.openmrs.contrib.qaframework.page;
+
+import org.openmrs.contrib.qaframework.helper.Page;
+import org.openqa.selenium.By;
+import org.openqa.selenium.Keys;
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.support.ui.ExpectedConditions;
+
+public class EncounterFormPage extends Page {
+
+	private static final String URL = "/admin/encounters/encounter.form";
+	private static final By PATIENT_FIELD= By.id("patientId_id_selection");
+	private static final By FIRST_AUTOCOMPLETE_RESULT = By.cssSelector("body ul li a span");
+	private static final By ENCOUNTER_DATE_TIME_FIELD= By.xpath("//input[@name='encounterDatetime']");
+	private static final By ROLE_SELECT = By.id("roleIds[0]");
+	private static final By PROVIDER_NAME_FIELD = By.name("providerName");
+	private static final By SAVE_ENCOUNTER_BUTTON = By.id("saveEncounterButton");
+	private static final By DELETE_CHECKBOX = By.id("voided");
+	private static final By VOID_REASON_FIELD = By.name("voidReason");
+	private static final By CLOSE_CALENDER_BUTTON = By.xpath("//button[@class='ui-datepicker-close ui-state-default ui-priority-primary ui-corner-all']");
+
+	public EncounterFormPage(Page page) {
+		super(page);
+	}
+
+	public void enterPatientIdentifier(String patientIdentifier) {
+		findElement(PATIENT_FIELD).sendKeys(patientIdentifier);
+		waitForElement(FIRST_AUTOCOMPLETE_RESULT);
+		clickOn(FIRST_AUTOCOMPLETE_RESULT);
+	}
+
+	public void enterEncounterDateTime(String encounterDateTime) {
+		waiter.until(ExpectedConditions.elementToBeClickable(ENCOUNTER_DATE_TIME_FIELD));
+		clickOn(ENCOUNTER_DATE_TIME_FIELD);
```

Comment:
is the filed for Date or time?

Reply:
> Is the filed for Date or time?

@kdaud it is a dateTime field.

Reply:
Could you share a screen shot of the filed?

Reply:
> Could you share a screen shot of the filed?

@kdaud  alright. It's label is Encounter Date.
![DateTimeField](https://user-images.githubusercontent.com/24396590/158327428-96e1208d-1da5-4a87-a15f-f7c5150e5ddf.png)

Reply:
How about renaming it to `ENCOUNTER_DATE_AND_TIME FILED`?

Reply:
> How about renaming it to `ENCOUNTER_DATE_AND_TIME FILED`?

@kdaud since dateTime is valid type for an input field, I would suggest we leave it as ENCOUNTER_DATE_TIME_FIELD.

Reply:
There are two variables here; Date, and Time so they need to be distinguished from within the variable name. Using ENCOUNTER_DATE_TIME_FILED suggests there is one input so that is why I find  ENCOUNTER_DATE_AND_TIME_FILED a better element naming.

Reply:
> There are two variables here; Date, and Time so they need to be distinguished from within the variable name. Using ENCOUNTER_DATE_TIME_FILED suggests there is one input so that is why I find ENCOUNTER_DATE_AND_TIME_FILED a better element naming.

@kdaud I understand that we are capturing Date and Time, however what is enabling use fill this data is a one  dateTime field. I see don't see the reason why we would name the variable ENCOUNTER_DATE_AND_TIME_FIELD.

Reply:
@kdaud to me if we call it ENCOUNTER_DATE_AND_TIME_FIELD  it will appear as if  we are combining two fields in one variable, one for date and the other for time.